DuckDuckGo/AuthenticationViewController.swift

```diff
@@ -58,18 +58,37 @@ class AuthenticationViewController: UIViewController {
             onCouldNotAuthenticate()
         }
     }
+    
+    class Authenticator {
+        func canAuthenticate() -> Bool {
+            return true
+        }
+
+        func authenticate(reason: String) async throws -> Bool {
+            
+            return true
+        }
+    }
 
+// Added Async/Await
     private func authenticate() {
         hideUnlockInstructions()
-        authenticator.authenticate { (success, _) in
-            if success {
-                self.onAuthenticationSucceeded()
-            } else {
-                self.onAuthenticationFailed()
+        Task {
+            do {
+                let success = try await authenticator.authenticate(reason: "Authentication required")
```

Comment:
The reason is hardcoded in English, but we localize our app into other languages, so it should come from the `UserText` file instead.